modules/apps/account/account-service/src/main/java/com/liferay/account/internal/role/AccountRoleManagerImpl.java

```diff
@@ -51,6 +53,34 @@ public AccountRole addAccountRole(
 		return new AccountRoleImpl(role);
 	}
 
+	@Override
+	public void addAccountRoleUser(
+			long accountEntryId, long roleId, long userId)
+		throws PortalException {
+
+		AccountEntry accountEntry = _accountEntryLocalService.getAccountEntry(
+			accountEntryId);
+
+		_userGroupRoleLocalService.addUserGroupRoles(
```

Comment:
Is account membership required in order to be assigned an account role? Looks like there's no check for account membership when assigning account roles.

Reply:
Membership to an account is not required.  There is a use case where we assign a B2 user an account role. In that case they would not be an account member.